main.cc

```diff
@@ -1609,6 +1609,69 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 gossiper.local().unregister_(mm.local().shared_from_this()).get();
             });
 
+            supervisor::notify("starting auth service");
```

Comment:
I don't see a comment that explains why the reordering it okay.

Reply:
+1, we need an explanation of what exactly all this code movement achieves.